test/driver-test/driver-test.cpp

```diff
@@ -0,0 +1,298 @@
+	expected_options = template_options;
+	expected_options.header = true;
+	expected_options.fom = file_open_mode::overwrite;
+	expected_options.prm = pass_report_mode::none;
+	expected_options.output_filepath = "array_test.out";
+	test_cmd_line(cmd_line, expected_options, "cmd-line 21");
+
+	cmd_line = "C:/stl-lite/array_test.exe -h no -s no -p detail -t 2 -fa a_file -run a_test_suite";
+	expected_options = template_options;
+	expected_options.header = false;
+	expected_options.header_text = "";
+	expected_options.summary = false;
+	expected_options.prm = pass_report_mode::detail;
+	expected_options.fail_threshold = 2;
+	expected_options.fom = file_open_mode::append;
+	expected_options.output_filepath = "a_file.out";
+	expected_options.suites_to_run = "a_test_suite";
+	test_cmd_line(cmd_line, expected_options, "cmd-line 22");
+
+	//multiple suites as values for -run
+	cmd_line = "C:/stl-lite/array_test.exe -run array_test;string_view_test;driver_test";
+	expected_options = template_options;
+	expected_options.suites_to_run = "array_test;string_view_test;driver_test";
+	test_cmd_line(cmd_line, expected_options, "cmd-line 23");
+
+	//every option specified
+	cmd_line = 
+		"C:/stl-lite/array_test.exe -h yes -ht Running_$cmd -s yes -p detail -t 2 -fa a_file -run tester";
+	expected_options = template_options;
+	expected_options.header = true;
+	expected_options.header_text = "Running_array_test";
+	expected_options.summary = true;
+	expected_options.prm = pass_report_mode::detail;
+	expected_options.fail_threshold = 2;
+	expected_options.fom = file_open_mode::append;
+	expected_options.output_filepath = "a_file.out";
+	expected_options.suites_to_run = "tester";
+	test_cmd_line(cmd_line, expected_options, "cmd-line 24");
+
+	cmd_line =
+		"C:/stl-lite/array_test.exe -h yes -ht Running_$cmd -s no -p indicate -t 10 -fo a_file -run tester";
+	expected_options = template_options;
+	expected_options.header = true;
+	expected_options.header_text = "Running_array_test";
+	expected_options.summary = false;
+	expected_options.prm = pass_report_mode::indicate;
+	expected_options.fail_threshold = 10;
+	expected_options.fom = file_open_mode::overwrite;
+	expected_options.output_filepath = "a_file.out";
+	expected_options.suites_to_run = "tester";
+	test_cmd_line(cmd_line, expected_options, "cmd-line 25");
+
+	cmd_line =
+		"C:/stl-lite/array_test.exe -h no -ht $cmd -s yes -p none -t 2 -fn a_file -run tester";
+	expected_options = template_options;
+	expected_options.header = false;
+	expected_options.header_text = "";
+	expected_options.summary = true;
+	expected_options.prm = pass_report_mode::none;
+	expected_options.fail_threshold = 2;
+	expected_options.fom = file_open_mode::new_file;
+	expected_options.output_filepath = "a_file.out";
+	expected_options.suites_to_run = "tester";
+	test_cmd_line(cmd_line, expected_options, "cmd-line 26");
+
+	//edge case: fail threshold
+	cmd_line = "C:/stl-lite/array_test.exe -t 65535"; //USHRT_MAX: compiler specific
```

Comment:
Use the macro I snuck into `get_fail_threshold` in `options.cpp`. It exists precisely to deal with data-type related issues.

```suggestion
	cmd_line = "C:/stl-lite/array_test.exe -t max"
```

Reply:
Thanks for the suggestion @smurthys, I had forgotten about that macro.